#152796595 Convert from Grunt to Gulp taskrunner

monsieurkayode opened the pull request:
#### What does this PR do?
Have Gulp as task runner
#### Description of Task to be completed?
* Setup app to use gulp task runner
* Add gulp file and configure tasks
* Delete grunt file and associated dependencies
#### How should this be manually tested?
* clone repo to local machine by running `git clone https://github.com/andela/elendil-cfh.git` in terminal or download zip
* cd into directory `elendil-cfh`
* install app dependencies by running `npm install`
* install and setup `mongodb` database
* create `.env` file, refer to `.env.example`
* run `npm start` in terminal
* navigate to `localhost:3000` on your browser
* run `npm test` to run tests
#### Any background context you want to provide?
None
#### What are the relevant pivotal tracker stories?
#152796595
#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/26303683/33093515-7401062e-cefd-11e7-88c5-474472b67853.png)

larrystone commented:
Great job bro
However, please fix the hound CI issues and the deployed app gave us a blank screen with this error on the console
`Uncaught Error: No module: ui.bootstrap`
please fix that too.
Thanks
<img width="1680" alt="screen shot 2017-11-22 at 8 16 36 am" src="https://user-images.githubusercontent.com/8871565/33114944-699a89a0-cf5f-11e7-9153-e1415a8fcf29.png">

monsieurkayode committed:
chore(gulp): convert from grunt to gulp taskrunner
- create and add gulp file
- install gulp and gulp plugins as core dependencies
- setup gulp task runner for compiling scss
- setup gulp task runner for linting
- setup gulp task runner for running tests
- setup gulp task runner for installing bower components
- setup gulp task runner to watch for changes and restart server
- modify scripts in package.json to run gulp tasks
- remove grunt and grunt plugins packages from dependencies
- delete gruntfile
- setup gulp task for installing and building bower components
- change path to bootsrap assets in app head and foot views template
- fix deployment on heroku using a gulp build task runner
- add a post build script to run after deployment build on heroku
[Finishes #152796595]

monsieurkayode committed:
Merge branch 'staging' into chore/152796595/convert-grunt-to-gulp

monsieurkayode committed:
Merge branch 'staging' into chore/152796595/convert-grunt-to-gulp
